modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/hash/PageMemoryHashIndexStorage.java

```diff
@@ -105,6 +108,8 @@ public PageMemoryHashIndexStorage(
         lowestRowId = RowId.lowestRowId(partitionId);
 
         highestRowId = RowId.highestRowId(partitionId);
+
+        lastBuiltIndexRow = readLastBuiltRowId();
```

Comment:
`IndexMeta` could have been passed into the constructor, because you already read it. This way you wouldn't need to go to the tree second time.

Reply:
But it could be `null` if the index is new, and when we create the **Hash/SortedIndexTree**, we save the new `IndexMeta` to the `IndexMetaTree`.

Reply:
Where is the code that creates the meta? Is it executed before the constructor? If the answer is "yes", then you do have the instance of the meta. If no, then where do you get the instance of the tree?

Reply:
Fix it